tests/basic/legacy_shared_sharedv4_service_migration_test.go

```diff
@@ -0,0 +1,713 @@
+			dash.VerifyFatal(err, nil, "Validate node rejoin init")
+			var rejoinedNode *api.StorageNode
+			t := func() (interface{}, bool, error) {
+				drvNodes, err := Inst().V.GetDriverNodes()
+				if err != nil {
+					return false, true, err
+				}
+
+				for _, n := range drvNodes {
+					if n.Hostname == pxNode.Hostname {
+						rejoinedNode = n
+						return true, false, nil
+					}
+				}
+
+				return false, true, fmt.Errorf("node %s not joined yet", pxNode.Name)
+			}
+			_, err = task.DoRetryWithTimeout(t, 15*time.Minute, defaultRetryInterval)
+			log.FailOnError(err, fmt.Sprintf("error joining the node [%s]", pxNode.Name))
+			dash.VerifyFatal(rejoinedNode != nil, true, fmt.Sprintf("verify node [%s] rejoined PX cluster", pxNode.Name))
+			err = Inst().S.RefreshNodeRegistry()
+			log.FailOnError(err, "error refreshing node registry")
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing storage drive endpoints")
+			decommissionedNode := node.Node{}
+			for _, n := range node.GetStorageDriverNodes() {
+				if n.Name == rejoinedNode.Hostname {
+					decommissionedNode = n
+					break
+				}
+			}
+			if decommissionedNode.Name == "" {
+				log.FailOnError(fmt.Errorf("rejoined node not found"), fmt.Sprintf("node [%s] not found in the node registry", rejoinedNode.Hostname))
+			}
+			err = Inst().V.WaitDriverUpOnNode(decommissionedNode, Inst().DriverStartTimeout)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Validate driver up on rejoined node [%s] after rejoining", decommissionedNode.Name))
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts, testrailID, runID)
+		DestroyApps(contexts, nil)
+	})
+})
+
+var _ = Describe("{LegacySharedToSharedv4ServiceRestartCoordinator}", func() {
+	var testrailID = 296732
+	var runID int
+	JustBeforeEach(func() {
+		StartTorpedoTest("LegacySharedVolumeAppRestartCoordinator", "Legacy Shared to Sharedv4 Service Migration and coordinator restart", nil, testrailID)
+		namespacePrefix := "lstsv4m-px-restart"
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+		setCreateLegacySharedAsSharedv4Service(false)
+		setMigrateLegacySharedToSharedv4Service(false)
+		contexts = make([]*scheduler.Context, 0)
+		numberNameSpaces := Inst().GlobalScaleFactor
+		if numberNameSpaces < numApps {
+			numberNameSpaces = numApps
+		}
+		for i := 0; i < numberNameSpaces; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("%s-%d", namespacePrefix, i))...)
+		}
+		// TODO: Skip non legacy shared tests
+		ValidateApplications(contexts)
+	})
+
+	ItLog := "Start Migration and after 2 minutes restart volume coordinator node"
+	It(ItLog, func() {
+		podMap := make(map[types.UID]bool)
+		volMap := make(map[string]bool)
+		for _, ctx := range contexts {
+			returnMapOfPodsUsingApiSharedVolumes(podMap, volMap, ctx)
+		}
+		var nodeForPxRestart *node.Node
+		for _, ctx := range contexts {
+			_, apiVol, attachedNode := getLegacySharedTestAppVol(ctx)
```

Comment:
here you are testing for only one context is that expected?

Reply:
es, we are shutting down coordinator for one context volume. If we have multiple volumes, coordinator for different volumes will be different, we want to restart only one node